web/app/js/src/jsoneditor-loader.js

```diff
@@ -51,22 +52,45 @@ $(document).on('click', '.jsonLoader', function(){
 });
 
 $(document).on('click', '.jsonSaver', function(){
-  var formData = new FormData();
-  var blob = new Blob([JSON.stringify(editor.get())], { type: 'application/json' });
-  formData.append('replace', true);
-  formData.append('upload_id', currentFileItemID);
-  formData.append('id', currentFileUploadID);
-  formData.append('type', 'experiments');
-  formData.append('file', blob);
-
-  $.ajax({
-    url: 'app/controllers/EntityController.php',
-    data: formData,
-    processData: false,
-    contentType: false,
-    type: 'POST',
-    success:function(){
-      $('.jsonSaver').attr('disabled', 1).text('Saved').css('cursor','default');
+  if(typeof currentFileUploadID === 'undefined'){
+    const realName = prompt('Enter name of the file');
+    if (realName == null) {
+      return;
     }
-  });
+    id = $('#main_form').find('input[name="id"]').attr('value');
+    $.post('app/controllers/EntityAjaxController.php', {
+      addFromString: true,
+      type: 'experiments',
+      id: id,
+      realName: realName,
+      fileType: 'json',
+      string: JSON.stringify(editor.get())
+    }).done(function(json) {
+      $('.jsonSaver').attr('disabled', 1).text('Saved').css('cursor','default');
+      $('#filesdiv').load('experiments.php?mode=edit&id=' + id + ' #filesdiv', function() {
+        makeEditableFileComment();
+      });
+      notif(json);
+    });
+  }
+  else{
```

Comment:
formatting issue here

Reply:
You mean just that "else{" part? So I should make it "else {"?

Reply:
I mean it should be `} else {` on the same line ;)

Reply:
Done :)